libvast/src/system/index.cpp

```diff
@@ -208,49 +208,152 @@ filesystem_actor& partition_factory::filesystem() {
 
 partition_actor partition_factory::operator()(const uuid& id) const {
   // Load partition from disk.
-  VAST_ASSERT(std::find(state_.persisted_partitions.begin(),
-                        state_.persisted_partitions.end(), id)
-              != state_.persisted_partitions.end());
+  if (std::find(state_.persisted_partitions.begin(),
+                state_.persisted_partitions.end(), id)
+      == state_.persisted_partitions.end()) {
+    VAST_WARN("{} can't load unknown partition {}", *state_.self, id);
+    // FIXME: returning a null actor crashes soon after.
+    // return {};
```

Comment:
This sounds like it still needs to be fixed

Reply:
I just tried this again and the issue has been resolved in the meantime. Instead of the commented out return I changed the warning message because it is more robust this way.